FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -18,7 +18,59 @@ public final class RemoteFeedLoader: FeedLoader {
 		self.client = client
 	}
 
+	private static let OK_200 = 200
+
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+				guard response.statusCode == RemoteFeedLoader.OK_200,
+				      let feedImages = RemoteFeedLoader.map(data) else {
+					return completion(.failure(Error.invalidData))
+				}
+
+				completion(.success(feedImages))
+			default:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	private static func map(_ data: Data) -> [FeedImage]? {
+		var feedImages: [FeedImage] = []
+		do {
+			let items = try JSONDecoder().decode(Items.self, from: data)
+			feedImages = items.map()
+		} catch {
+			return nil
+		}
+
+		return feedImages
+	}
+
+	private final class Items: Decodable {
+		fileprivate let items: [FeedAPIImage]
+
+		fileprivate func map() -> [FeedImage] {
```

Comment:
`fileprivate` is not needed here since the class is already private and other files can't access it. You can remove the fileprivate access level.

```suggestion
		func map() -> [FeedImage] {
```